cmd/swarmctl/service/flagparser/flags.go

```diff
@@ -32,6 +32,7 @@ func AddServiceFlags(flags *pflag.FlagSet) {
 	flags.String("memory-limit", "", "memory limit (e.g. 512m)")
 	flags.String("cpu-reservation", "", "number of CPU cores reserved (e.g. 0.5)")
 	flags.String("cpu-limit", "", "CPU cores limit (e.g. 0.5)")
+	flags.String("generic-service-resource", "", "user defined resources request (e.g. gpu=3;fpga=1)")
```

Comment:
nit: this is a service command, have service in the flag isn't necessary (can be shortened to `--generic-resource`)